File: NpvCalculator-Angular/src/app/_models/net-present-value-request.ts
import { PeriodAmount } from './period-amount';

export interface NetPresentValueRequest {
    netPresentValueId: number;
    name: string;
    initialInvestment: number;
    lowerBoundDiscountRate: number;
    upperBoundDiscountRate: number;
    discountRateIncrement: number;
    cashFlows: PeriodAmount[];
}

File: NpvCalculator.Core/Classes/NetPresentValueRequest.cs
using System.Collections.Generic;

namespace NpvCalculator.Core.Classes
{
    public class NetPresentValueRequest
    {
        public int NetPresentValueId { get; set; }
        public string Name { get; set; }
        public double InitialInvestment { get; set; }
        public double LowerBoundDiscountRate { get; set; }
        public double UpperBoundDiscountRate { get; set; }
        public double DiscountRateIncrement { get; set; }
        public IEnumerable<PeriodAmount> CashFlows { get; set; }
        public bool SaveToDatabase { get; set; }
    }
}
